### ISSUE TITLE
Terrain Editor

### ISSUE THREAD

**Issue by MR-Alex42** [Frooxius/NeosPublic#118]
Currently Neos is currently missing tools for editing terrains. 
There were already some options explored in the 'Neos The Origin' demo
but there should be more options available like heighten, lowering and cutting the geometry.

Maybe a voxel based approach for this would would be best as this not just would allow creating terrains but all kind objects as well.

**Comment by Frooxius** [comment]
Yeah this is planned, but requires some non-trivial additions to the data model to be implemented efficiently. I will likely provide both systems - height based and voxel based, since they each have their applications and benefits.

I'm thinking about some potential hybrid system, which would let you mix and match the two as well, but I'll need to give that more thought.

The goal is to base this around some new data primitives / elements, which could be used for variety of other purposes as well.
